R/PackageInheritanceReporter.R

```diff
@@ -0,0 +1,212 @@
+#' @importFrom data.table data.table rbindlist
+#' @importFrom methods is
+#' @export
+InheritanceReporter <- R6::R6Class(
+    "InheritanceReporter",
+    inherit = AbstractGraphReporter,
+
+    public = list(
+        get_summary_view = function(){
+
+            # Calculate network measures if not already done
+            # since we want the node measures in summary
+            invisible(self$network_measures)
+
+            # Create DT for display
+            tableObj <- DT::datatable(
+                data = self$nodes
+                , rownames = FALSE
+                , options = list(
+                    searching = FALSE
+                    , pageLength = 50
+                    , lengthChange = FALSE
+                )
+            )
+            # Round the double columns to three digits for formatting reasons
+            numCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
+            tableObj <- DT::formatRound(columns = numCols, table = tableObj
+                                        , digits=3)
+            return(tableObj)
+        }
+    ),
+
+    active = list(
+        nodes = function(){
+            if (is.null(private$cache$nodes)){
+                log_info("Extracting classes as nodes...")
+
+                pkg_env <- private$get_pkg_env()
+
+                nodeList <- list(data.table::data.table(
+                    node = character(0), classType = character(0)
+                ))
+
+                for (thisObjName in names(pkg_env)) {
+
+                    thisObj <- get(thisObjName, pkg_env)
+
+                    # S4 classes and References classes
+                    # Specified class name is the important name
+                    if (grepl("^\\.__C__", thisObjName) & isS4(thisObj)) {
+
+                        if (methods::is(thisObj, "refClassRepresentation")) {
+                            thisObjClassType <- "Reference"
+                        } else {
+                            thisObjClassType <- "S4"
+                        }
+
+                        nodeList <- c(nodeList, list(data.table::data.table(
+                            node = thisObj@className
+                            , classType = thisObjClassType
+                        )))
+
+                    # R6 classes
+                    # Generator object name is the important name
+                    } else if (R6::is.R6Class(thisObj)) {
+
+                        nodeList <- c(nodeList, list(data.table::data.table(
+                            node = thisObjName
+                            , classType = "R6"
+                        )))
+
+                    }
+
+                }
+
+                nodeDT <- data.table::rbindlist(nodeList)
+                if (nrow(nodeDT) == 0) {
+                    msg <- sprintf(
+                        'No Reference Class or R6 Class definitions found in package %s'
+                        , self$pkg_name
+                    )
+                    log_warn(msg)
```

Comment:
I like the choice of warning here. At first I was like "that feels aggressive", but if someone is using `InheritanceReporter` then I'd think they would want to know that they used something intended for classes on a package with no classes.

If we were putting this in `DefaultReporters`, I would ask to change this to `log_info` maybe